file save triggers error

JabRef 5.1--2020-05-08--d18ce55
Linux 5.6.8-200.fc31.x86_64 amd64 
Java 14.0.1

<!-- Add a clear and concise description of what the bug is. -->

When I edit text in the entry editor, I often get this exception, without further action. I suspected it might be related to autocompletion, but I read @Siedlerchr as this not being the case at https://github.com/JabRef/jabref/issues/6351. As you see at https://github.com/JabRef/jabref/issues/6351, I routinely faced this error for for 8 days now.

The bug issue heading might be equivalent to https://github.com/JabRef/jabref/issues/6102? - but I am not sure whether it is the same issue.

How to reproduce: edit any text field and after a few seconds the exception shows up. 



<details>
  <summary>Log File</summary>
  
java.lang.IllegalStateException: Task must only be used from the FX Application Thread
	at org.jabref.merged.module/javafx.concurrent.Task.checkThread(Unknown Source)
	at org.jabref.merged.module/javafx.concurrent.Task.isRunning(Unknown Source)
	at org.jabref.merged.module/org.controlsfx.control.textfield.AutoCompletionBinding.onUserInputChanged(Unknown Source)
	at org.jabref.merged.module/org.controlsfx.control.textfield.AutoCompletionBinding.setUserInput(Unknown Source)
	at org.jabref/org.jabref.gui.autocompleter.AutoCompletionTextInputBinding.setUserInputText(Unknown Source)
	at org.jabref/org.jabref.gui.autocompleter.AutoCompletionTextInputBinding.lambda$new$0(Unknown Source)
	at org.jabref.merged.module/com.sun.javafx.binding.ExpressionHelper$Generic.fireValueChangedEvent(Unknown Source)
	at org.jabref.merged.module/com.sun.javafx.binding.ExpressionHelper.fireValueChangedEvent(Unknown Source)
	at org.jabref.merged.module/javafx.scene.control.TextInputControl$TextProperty.fireValueChangedEvent(Unknown Source)
	at org.jabref.merged.module/javafx.scene.control.TextInputControl$TextProperty.markInvalid(Unknown Source)
	at org.jabref.merged.module/javafx.scene.control.TextInputControl$TextProperty.controlContentHasChanged(Unknown Source)
	at org.jabref.merged.module/javafx.scene.control.TextInputControl.lambda$new$0(Unknown Source)
	at org.jabref.merged.module/com.sun.javafx.binding.ExpressionHelper$SingleInvalidation.fireValueChangedEvent(Unknown Source)
	at org.jabref.merged.module/com.sun.javafx.binding.ExpressionHelper.fireValueChangedEvent(Unknown Source)
	at org.jabref.merged.module/javafx.scene.control.TextArea$TextAreaContent.insert(Unknown Source)
	at org.jabref.merged.module/javafx.scene.control.TextInputControl.replaceText(Unknown Source)
	at org.jabref.merged.module/javafx.scene.control.TextInputControl.filterAndSet(Unknown Source)
	at org.jabref.merged.module/javafx.scene.control.TextInputControl$TextProperty.doSet(Unknown Source)
	at org.jabref.merged.module/javafx.scene.control.TextInputControl$TextProperty.set(Unknown Source)
	at org.jabref.merged.module/javafx.scene.control.TextInputControl$TextProperty.set(Unknown Source)
	at org.jabref.merged.module/javafx.beans.property.StringProperty.setValue(Unknown Source)
	at org.jabref.merged.module/javafx.beans.property.StringProperty.setValue(Unknown Source)
	at org.jabref.merged.module/com.sun.javafx.binding.BidirectionalBinding$TypedGenericBidirectionalBinding.changed(Unknown Source)
	at org.jabref.merged.module/com.sun.javafx.binding.ExpressionHelper$Generic.fireValueChangedEvent(Unknown Source)
	at org.jabref.merged.module/com.sun.javafx.binding.ExpressionHelper.fireValueChangedEvent(Unknown Source)
	at org.jabref.merged.module/javafx.beans.property.StringPropertyBase.fireValueChangedEvent(Unknown Source)
	at org.jabref.merged.module/javafx.beans.property.StringPropertyBase.markInvalid(Unknown Source)
	at org.jabref.merged.module/javafx.beans.property.StringPropertyBase.set(Unknown Source)
	at org.jabref.merged.module/javafx.beans.property.StringPropertyBase.set(Unknown Source)
	at org.jabref.merged.module/javafx.beans.property.StringProperty.setValue(Unknown Source)
	at org.jabref.merged.module/javafx.beans.property.StringProperty.setValue(Unknown Source)
	at org.jabref/org.jabref.gui.util.BindingsHelper$BidirectionalBinding.updateLocked(Unknown Source)
	at org.jabref/org.jabref.gui.util.BindingsHelper$BidirectionalBinding.changedB(Unknown Source)
	at org.jabref.merged.module/com.sun.javafx.binding.ExpressionHelper$SingleChange.fireValueChangedEvent(Unknown Source)
	at org.jabref.merged.module/com.sun.javafx.binding.ExpressionHelper.fireValueChangedEvent(Unknown Source)
	at org.jabref.merged.module/javafx.beans.binding.ObjectBinding.invalidate(Unknown Source)
	at org.jabref.merged.module/com.sun.javafx.binding.BindingHelperObserver.invalidated(Unknown Source)
	at org.jabref.merged.module/com.sun.javafx.collections.MapListenerHelper$Generic.fireValueChangedEvent(Unknown Source)
	at org.jabref.merged.module/com.sun.javafx.collections.MapListenerHelper.fireValueChangedEvent(Unknown Source)
	at org.jabref.merged.module/com.sun.javafx.collections.ObservableMapWrapper.callObservers(Unknown Source)
	at org.jabref.merged.module/com.sun.javafx.collections.ObservableMapWrapper.put(Unknown Source)
	at org.jabref/org.jabref.model.entry.BibEntry.setField(Unknown Source)
	at org.jabref/org.jabref.model.cleanup.FieldFormatterCleanup.cleanupSingleField(Unknown Source)
	at org.jabref/org.jabref.model.cleanup.FieldFormatterCleanup.cleanupAllFields(Unknown Source)
	at org.jabref/org.jabref.model.cleanup.FieldFormatterCleanup.cleanup(Unknown Source)
	at org.jabref/org.jabref.logic.exporter.BibDatabaseWriter.applySaveActions(Unknown Source)
	at org.jabref/org.jabref.logic.exporter.BibDatabaseWriter.savePartOfDatabase(Unknown Source)
	at org.jabref/org.jabref.logic.exporter.BibDatabaseWriter.saveDatabase(Unknown Source)
	at org.jabref/org.jabref.gui.exporter.SaveDatabaseAction.saveDatabase(Unknown Source)
	at org.jabref/org.jabref.gui.exporter.SaveDatabaseAction.save(Unknown Source)
	at org.jabref/org.jabref.gui.exporter.SaveDatabaseAction.save(Unknown Source)
	at org.jabref/org.jabref.gui.exporter.SaveDatabaseAction.save(Unknown Source)
	at org.jabref/org.jabref.gui.dialogs.AutosaveUiManager.listen(Unknown Source)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
	at org.jabref.merged.module/com.google.common.eventbus.Subscriber.invokeSubscriberMethod(Unknown Source)
	at org.jabref.merged.module/com.google.common.eventbus.Subscriber$SynchronizedSubscriber.invokeSubscriberMethod(Unknown Source)
	at org.jabref.merged.module/com.google.common.eventbus.Subscriber$1.run(Unknown Source)
	at org.jabref.merged.module/com.google.common.util.concurrent.DirectExecutor.execute(Unknown Source)
	at org.jabref.merged.module/com.google.common.eventbus.Subscriber.dispatchEvent(Unknown Source)
	at org.jabref.merged.module/com.google.common.eventbus.Dispatcher$PerThreadQueuedDispatcher.dispatch(Unknown Source)
	at org.jabref.merged.module/com.google.common.eventbus.EventBus.post(Unknown Source)
	at org.jabref/org.jabref.logic.autosaveandbackup.AutosaveManager.lambda$listen$0(Unknown Source)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)
	at java.base/java.util.concurrent.FutureTask.run(Unknown Source)
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.base/java.lang.Thread.run(Unknown Source)
</details>

yeah, but got this now after saving and writing this line:
<details>
  <summary>Log File</summary>
  


org.jabref.logic.exporter.SaveException: Problems saving: java.nio.file.NoSuchFileException: /home/ilippert/Documents/Studies/Publications&Products/Resources/Literature-bibtex.bib.tmp -> /home/ilippert/Documents/Studies/Publications&Products/Resources/Literature-bibtex.bib
	at org.jabref/org.jabref.gui.exporter.SaveDatabaseAction.saveDatabase(SaveDatabaseAction.java:239)
	at org.jabref/org.jabref.gui.exporter.SaveDatabaseAction.save(SaveDatabaseAction.java:195)
	at org.jabref/org.jabref.gui.exporter.SaveDatabaseAction.save(SaveDatabaseAction.java:177)
	at org.jabref/org.jabref.gui.exporter.SaveDatabaseAction.save(SaveDatabaseAction.java:71)
	at org.jabref/org.jabref.gui.exporter.SaveAction.execute(SaveAction.java:39)
	at org.jabref/org.jabref.gui.actions.JabRefAction.lambda$new$2(JabRefAction.java:40)
	at org.jabref.merged.module/org.controlsfx.control.action.Action.handle(Action.java:419)
	at org.jabref.merged.module/org.controlsfx.control.action.Action.handle(Action.java:64)
	at org.jabref.merged.module/com.sun.javafx.event.CompositeEventHandler.dispatchBubblingEvent(CompositeEventHandler.java:86)
	at org.jabref.merged.module/com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:234)
	at org.jabref.merged.module/com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:191)
	at org.jabref.merged.module/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:58)
	at org.jabref.merged.module/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at org.jabref.merged.module/com.sun.javafx.event.EventUtil.fireEventImpl(EventUtil.java:74)
	at org.jabref.merged.module/com.sun.javafx.event.EventUtil.fireEvent(EventUtil.java:49)
	at org.jabref.merged.module/javafx.event.Event.fireEvent(Event.java:198)
	at org.jabref.merged.module/javafx.scene.control.MenuItem.fire(MenuItem.java:459)
	at org.jabref.merged.module/com.sun.javafx.scene.control.ControlAcceleratorSupport.lambda$doAcceleratorInstall$1(ControlAcceleratorSupport.java:165)
	at org.jabref.merged.module/com.sun.javafx.scene.KeyboardShortcutsHandler.processAccelerators(KeyboardShortcutsHandler.java:382)
	at org.jabref.merged.module/com.sun.javafx.scene.KeyboardShortcutsHandler.dispatchBubblingEvent(KeyboardShortcutsHandler.java:162)
	at org.jabref.merged.module/com.sun.javafx.event.CompositeEventDispatcher.dispatchBubblingEvent(CompositeEventDispatcher.java:59)
	at org.jabref.merged.module/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:58)
	at org.jabref.merged.module/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at org.jabref.merged.module/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
	at org.jabref.merged.module/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at org.jabref.merged.module/com.sun.javafx.event.EventUtil.fireEventImpl(EventUtil.java:74)
	at org.jabref.merged.module/com.sun.javafx.event.EventUtil.fireEvent(EventUtil.java:54)
	at org.jabref.merged.module/javafx.event.Event.fireEvent(Event.java:198)
	at org.jabref.merged.module/javafx.scene.Scene$KeyHandler.process(Scene.java:4098)
	at org.jabref.merged.module/javafx.scene.Scene.processKeyEvent(Scene.java:2157)
	at org.jabref.merged.module/javafx.scene.Scene$ScenePeerListener.keyEvent(Scene.java:2625)
	at org.jabref.merged.module/com.sun.javafx.tk.quantum.GlassViewEventHandler$KeyEventNotification.run(GlassViewEventHandler.java:217)
	at org.jabref.merged.module/com.sun.javafx.tk.quantum.GlassViewEventHandler$KeyEventNotification.run(GlassViewEventHandler.java:149)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:391)
	at org.jabref.merged.module/com.sun.javafx.tk.quantum.GlassViewEventHandler.lambda$handleKeyEvent$1(GlassViewEventHandler.java:248)
	at org.jabref.merged.module/com.sun.javafx.tk.quantum.QuantumToolkit.runWithoutRenderLock(QuantumToolkit.java:412)
	at org.jabref.merged.module/com.sun.javafx.tk.quantum.GlassViewEventHandler.handleKeyEvent(GlassViewEventHandler.java:247)
	at org.jabref.merged.module/com.sun.glass.ui.View.handleKeyEvent(View.java:547)
	at org.jabref.merged.module/com.sun.glass.ui.View.notifyKey(View.java:971)
	at org.jabref.merged.module/com.sun.glass.ui.gtk.GtkApplication._runLoop(Native Method)
	at org.jabref.merged.module/com.sun.glass.ui.gtk.GtkApplication.lambda$runLoop$11(GtkApplication.java:277)
	at java.base/java.lang.Thread.run(Thread.java:832)
Caused by: java.nio.file.NoSuchFileException: /home/ilippert/Documents/Studies/Publications&Products/Resources/Literature-bibtex.bib.tmp -> /home/ilippert/Documents/Studies/Publications&Products/Resources/Literature-bibtex.bib
	at java.base/sun.nio.fs.UnixException.translateToIOException(UnixException.java:92)
	at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:111)
	at java.base/sun.nio.fs.UnixCopyFile.move(UnixCopyFile.java:415)
	at java.base/sun.nio.fs.UnixFileSystemProvider.move(UnixFileSystemProvider.java:267)
	at java.base/java.nio.file.Files.move(Files.java:1426)
	at org.jabref/org.jabref.logic.exporter.AtomicFileOutputStream.close(AtomicFileOutputStream.java:196)
	at java.base/sun.nio.cs.StreamEncoder.implClose(StreamEncoder.java:353)
	at java.base/sun.nio.cs.StreamEncoder.close(StreamEncoder.java:168)
	at java.base/java.io.OutputStreamWriter.close(OutputStreamWriter.java:255)
	at org.jabref/org.jabref.logic.exporter.BibDatabaseWriter.savePartOfDatabase(BibDatabaseWriter.java:221)
	at org.jabref/org.jabref.logic.exporter.BibDatabaseWriter.saveDatabase(BibDatabaseWriter.java:161)
	at org.jabref/org.jabref.gui.exporter.SaveDatabaseAction.saveDatabase(SaveDatabaseAction.java:228)
	... 41 more

</details>

still on
JabRef 5.1-PullRequest6455.368--2020-05-09--fff4926
Linux 5.6.8-200.fc31.x86_64 amd64 
Java 14.0.1

Without me doing anything

<details>
  <summary>Log File</summary>
java.lang.RuntimeException: Bidirectional binding failed, setting to the previous value
	at org.jabref.merged.module/com.sun.javafx.binding.BidirectionalBinding$TypedGenericBidirectionalBinding.changed(BidirectionalBinding.java:619)
	at org.jabref.merged.module/com.sun.javafx.binding.ExpressionHelper$Generic.fireValueChangedEvent(ExpressionHelper.java:360)
	at org.jabref.merged.module/com.sun.javafx.binding.ExpressionHelper.fireValueChangedEvent(ExpressionHelper.java:80)
	at org.jabref.merged.module/javafx.beans.property.StringPropertyBase.fireValueChangedEvent(StringPropertyBase.java:104)
	at org.jabref.merged.module/javafx.beans.property.StringPropertyBase.markInvalid(StringPropertyBase.java:111)
	at org.jabref.merged.module/javafx.beans.property.StringPropertyBase.set(StringPropertyBase.java:145)
	at org.jabref.merged.module/javafx.beans.property.StringPropertyBase.set(StringPropertyBase.java:50)
	at org.jabref.merged.module/javafx.beans.property.StringProperty.setValue(StringProperty.java:65)
	at org.jabref.merged.module/javafx.beans.property.StringProperty.setValue(StringProperty.java:57)
	at org.jabref/org.jabref.gui.util.BindingsHelper$BidirectionalBinding.updateLocked(BindingsHelper.java:254)
	at org.jabref/org.jabref.gui.util.BindingsHelper$BidirectionalBinding.changedB(BindingsHelper.java:247)
	at org.jabref.merged.module/com.sun.javafx.binding.ExpressionHelper$SingleChange.fireValueChangedEvent(ExpressionHelper.java:181)
	at org.jabref.merged.module/com.sun.javafx.binding.ExpressionHelper.fireValueChangedEvent(ExpressionHelper.java:80)
	at org.jabref.merged.module/javafx.beans.binding.ObjectBinding.invalidate(ObjectBinding.java:170)
	at org.jabref.merged.module/com.sun.javafx.binding.BindingHelperObserver.invalidated(BindingHelperObserver.java:52)
	at org.jabref.merged.module/com.sun.javafx.collections.MapListenerHelper$Generic.fireValueChangedEvent(MapListenerHelper.java:320)
	at org.jabref.merged.module/com.sun.javafx.collections.MapListenerHelper.fireValueChangedEvent(MapListenerHelper.java:72)
	at org.jabref.merged.module/com.sun.javafx.collections.ObservableMapWrapper.callObservers(ObservableMapWrapper.java:115)
	at org.jabref.merged.module/com.sun.javafx.collections.ObservableMapWrapper.put(ObservableMapWrapper.java:169)
	at org.jabref/org.jabref.model.entry.BibEntry.setField(BibEntry.java:539)
	at org.jabref/org.jabref.model.cleanup.FieldFormatterCleanup.cleanupSingleField(FieldFormatterCleanup.java:66)
	at org.jabref/org.jabref.model.cleanup.FieldFormatterCleanup.cleanupAllFields(FieldFormatterCleanup.java:78)
	at org.jabref/org.jabref.model.cleanup.FieldFormatterCleanup.cleanup(FieldFormatterCleanup.java:33)
	at org.jabref/org.jabref.logic.exporter.BibDatabaseWriter.applySaveActions(BibDatabaseWriter.java:78)
	at org.jabref/org.jabref.logic.exporter.BibDatabaseWriter.savePartOfDatabase(BibDatabaseWriter.java:190)
	at org.jabref/org.jabref.logic.exporter.BibDatabaseWriter.saveDatabase(BibDatabaseWriter.java:161)
	at org.jabref/org.jabref.gui.exporter.SaveDatabaseAction.saveDatabase(SaveDatabaseAction.java:228)
	at org.jabref/org.jabref.gui.exporter.SaveDatabaseAction.save(SaveDatabaseAction.java:195)
	at org.jabref/org.jabref.gui.exporter.SaveDatabaseAction.save(SaveDatabaseAction.java:177)
	at org.jabref/org.jabref.gui.exporter.SaveDatabaseAction.save(SaveDatabaseAction.java:75)
	at org.jabref/org.jabref.gui.dialogs.AutosaveUiManager.listen(AutosaveUiManager.java:28)
	at jdk.internal.reflect.GeneratedMethodAccessor29.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:564)
	at org.jabref.merged.module/com.google.common.eventbus.Subscriber.invokeSubscriberMethod(Subscriber.java:87)
	at org.jabref.merged.module/com.google.common.eventbus.Subscriber$SynchronizedSubscriber.invokeSubscriberMethod(Subscriber.java:144)
	at org.jabref.merged.module/com.google.common.eventbus.Subscriber$1.run(Subscriber.java:72)
	at org.jabref.merged.module/com.google.common.util.concurrent.DirectExecutor.execute(DirectExecutor.java:30)
	at org.jabref.merged.module/com.google.common.eventbus.Subscriber.dispatchEvent(Subscriber.java:67)
	at org.jabref.merged.module/com.google.common.eventbus.Dispatcher$PerThreadQueuedDispatcher.dispatch(Dispatcher.java:108)
	at org.jabref.merged.module/com.google.common.eventbus.EventBus.post(EventBus.java:212)
	at org.jabref/org.jabref.logic.autosaveandbackup.AutosaveManager.lambda$listen$0(AutosaveManager.java:46)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630)
	at java.base/java.lang.Thread.run(Thread.java:832)
Caused by: java.lang.StringIndexOutOfBoundsException: index 515, length 515
	at java.base/java.lang.String.checkIndex(String.java:3715)
	at java.base/java.lang.AbstractStringBuilder.charAt(AbstractStringBuilder.java:362)
	at java.base/java.lang.StringBuilder.charAt(StringBuilder.java:85)
	at org.jabref.merged.module/javafx.scene.control.TextArea$TextAreaContent.get(TextArea.java:133)
	at org.jabref.merged.module/javafx.scene.control.TextArea$TextAreaContent.get(TextArea.java:311)
	at org.jabref.merged.module/javafx.scene.control.TextArea$TextAreaContent.get(TextArea.java:88)
	at org.jabref.merged.module/javafx.scene.control.TextInputControl$TextProperty.get(TextInputControl.java:1365)
	at org.jabref.merged.module/javafx.scene.control.TextInputControl$TextProperty.get(TextInputControl.java:1349)
	at org.jabref.merged.module/javafx.beans.binding.StringExpression.getValue(StringExpression.java:51)
	at org.jabref.merged.module/javafx.beans.binding.StringExpression.getValue(StringExpression.java:47)
	at org.jabref.merged.module/com.sun.javafx.binding.ExpressionHelper$Generic.fireValueChangedEvent(ExpressionHelper.java:355)
	at org.jabref.merged.module/com.sun.javafx.binding.ExpressionHelper.fireValueChangedEvent(ExpressionHelper.java:80)
	at org.jabref.merged.module/javafx.scene.control.TextInputControl$TextProperty.fireValueChangedEvent(TextInputControl.java:1438)
	at org.jabref.merged.module/javafx.scene.control.TextInputControl$TextProperty.markInvalid(TextInputControl.java:1442)
	at org.jabref.merged.module/javafx.scene.control.TextInputControl$TextProperty.set(TextInputControl.java:1373)
	at org.jabref.merged.module/javafx.scene.control.TextInputControl$TextProperty.set(TextInputControl.java:1349)
	at org.jabref.merged.module/javafx.beans.property.StringProperty.setValue(StringProperty.java:65)
	at org.jabref.merged.module/javafx.beans.property.StringProperty.setValue(StringProperty.java:57)
	at org.jabref.merged.module/com.sun.javafx.binding.BidirectionalBinding$TypedGenericBidirectionalBinding.changed(BidirectionalBinding.java:601)
	... 47 more
</details>

however, just before I read your question, I got this (I think similar to before)

<details>
  <summary>Log File</summary>
  
  ```
  org.jabref.logic.exporter.SaveException: Problems saving: java.nio.file.NoSuchFileException: /home/ilippert/Documents/Studies/Publications&Products/Resources/Literature-bibtex.bib.tmp -> /home/ilippert/Documents/Studies/Publications&Products/Resources/Literature-bibtex.bib
	at org.jabref/org.jabref.gui.exporter.SaveDatabaseAction.saveDatabase(SaveDatabaseAction.java:239)
	at org.jabref/org.jabref.gui.exporter.SaveDatabaseAction.save(SaveDatabaseAction.java:195)
	at org.jabref/org.jabref.gui.exporter.SaveDatabaseAction.save(SaveDatabaseAction.java:177)
	at org.jabref/org.jabref.gui.exporter.SaveDatabaseAction.save(SaveDatabaseAction.java:71)
	at org.jabref/org.jabref.gui.exporter.SaveAction.execute(SaveAction.java:39)
	at org.jabref/org.jabref.gui.actions.JabRefAction.lambda$new$2(JabRefAction.java:40)
	at org.jabref.merged.module/org.controlsfx.control.action.Action.handle(Action.java:419)
	at org.jabref.merged.module/org.controlsfx.control.action.Action.handle(Action.java:64)
	at org.jabref.merged.module/com.sun.javafx.event.CompositeEventHandler.dispatchBubblingEvent(CompositeEventHandler.java:86)
	at org.jabref.merged.module/com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:234)
	at org.jabref.merged.module/com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:191)
	at org.jabref.merged.module/com.sun.javafx.event.CompositeEventDispatcher.dispatchBubblingEvent(CompositeEventDispatcher.java:59)
	at org.jabref.merged.module/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:58)
	at org.jabref.merged.module/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at org.jabref.merged.module/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
	at org.jabref.merged.module/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at org.jabref.merged.module/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
	at org.jabref.merged.module/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at org.jabref.merged.module/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
	at org.jabref.merged.module/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at org.jabref.merged.module/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
	at org.jabref.merged.module/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at org.jabref.merged.module/com.sun.javafx.event.EventUtil.fireEventImpl(EventUtil.java:74)
	at org.jabref.merged.module/com.sun.javafx.event.EventUtil.fireEvent(EventUtil.java:49)
	at org.jabref.merged.module/javafx.event.Event.fireEvent(Event.java:198)
	at org.jabref.merged.module/javafx.scene.Node.fireEvent(Node.java:8885)
	at org.jabref.merged.module/javafx.scene.control.Button.fire(Button.java:203)
	at org.jabref.merged.module/com.sun.javafx.scene.control.behavior.ButtonBehavior.mouseReleased(ButtonBehavior.java:206)
	at org.jabref.merged.module/com.sun.javafx.scene.control.inputmap.InputMap.handle(InputMap.java:274)
	at org.jabref.merged.module/com.sun.javafx.event.CompositeEventHandler$NormalEventHandlerRecord.handleBubblingEvent(CompositeEventHandler.java:247)
	at org.jabref.merged.module/com.sun.javafx.event.CompositeEventHandler.dispatchBubblingEvent(CompositeEventHandler.java:80)
	at org.jabref.merged.module/com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:234)
	at org.jabref.merged.module/com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:191)
	at org.jabref.merged.module/com.sun.javafx.event.CompositeEventDispatcher.dispatchBubblingEvent(CompositeEventDispatcher.java:59)
	at org.jabref.merged.module/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:58)
	at org.jabref.merged.module/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at org.jabref.merged.module/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
	at org.jabref.merged.module/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at org.jabref.merged.module/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
	at org.jabref.merged.module/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at org.jabref.merged.module/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
	at org.jabref.merged.module/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at org.jabref.merged.module/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
	at org.jabref.merged.module/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at org.jabref.merged.module/com.sun.javafx.event.EventUtil.fireEventImpl(EventUtil.java:74)
	at org.jabref.merged.module/com.sun.javafx.event.EventUtil.fireEvent(EventUtil.java:54)
	at org.jabref.merged.module/javafx.event.Event.fireEvent(Event.java:198)
	at org.jabref.merged.module/javafx.scene.Scene$MouseHandler.process(Scene.java:3890)
	at org.jabref.merged.module/javafx.scene.Scene.processMouseEvent(Scene.java:1885)
	at org.jabref.merged.module/javafx.scene.Scene$ScenePeerListener.mouseEvent(Scene.java:2618)
	at org.jabref.merged.module/com.sun.javafx.tk.quantum.GlassViewEventHandler$MouseEventNotification.run(GlassViewEventHandler.java:409)
	at org.jabref.merged.module/com.sun.javafx.tk.quantum.GlassViewEventHandler$MouseEventNotification.run(GlassViewEventHandler.java:299)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:391)
	at org.jabref.merged.module/com.sun.javafx.tk.quantum.GlassViewEventHandler.lambda$handleMouseEvent$2(GlassViewEventHandler.java:447)
	at org.jabref.merged.module/com.sun.javafx.tk.quantum.QuantumToolkit.runWithoutRenderLock(QuantumToolkit.java:412)
	at org.jabref.merged.module/com.sun.javafx.tk.quantum.GlassViewEventHandler.handleMouseEvent(GlassViewEventHandler.java:446)
	at org.jabref.merged.module/com.sun.glass.ui.View.handleMouseEvent(View.java:556)
	at org.jabref.merged.module/com.sun.glass.ui.View.notifyMouse(View.java:942)
	at org.jabref.merged.module/com.sun.glass.ui.gtk.GtkApplication._runLoop(Native Method)
	at org.jabref.merged.module/com.sun.glass.ui.gtk.GtkApplication.lambda$runLoop$11(GtkApplication.java:277)
	at java.base/java.lang.Thread.run(Thread.java:832)
Caused by: java.nio.file.NoSuchFileException: /home/ilippert/Documents/Studies/Publications&Products/Resources/Literature-bibtex.bib.tmp -> /home/ilippert/Documents/Studies/Publications&Products/Resources/Literature-bibtex.bib
	at java.base/sun.nio.fs.UnixException.translateToIOException(UnixException.java:92)
	at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:111)
	at java.base/sun.nio.fs.UnixCopyFile.move(UnixCopyFile.java:415)
	at java.base/sun.nio.fs.UnixFileSystemProvider.move(UnixFileSystemProvider.java:267)
	at java.base/java.nio.file.Files.move(Files.java:1426)
	at org.jabref/org.jabref.logic.exporter.AtomicFileOutputStream.close(AtomicFileOutputStream.java:196)
	at java.base/sun.nio.cs.StreamEncoder.implClose(StreamEncoder.java:353)
	at java.base/sun.nio.cs.StreamEncoder.close(StreamEncoder.java:168)
	at java.base/java.io.OutputStreamWriter.close(OutputStreamWriter.java:255)
	at org.jabref/org.jabref.logic.exporter.BibDatabaseWriter.savePartOfDatabase(BibDatabaseWriter.java:221)
	at org.jabref/org.jabref.logic.exporter.BibDatabaseWriter.saveDatabase(BibDatabaseWriter.java:161)
	at org.jabref/org.jabref.gui.exporter.SaveDatabaseAction.saveDatabase(SaveDatabaseAction.java:228)
	... 60 more
  ```
</details>

Thanks, okay, this is a different error now which is related to our Backup-Manager. For some reason the tmp file cannot be renamed (moved) to the bib file, because it's not/no longer existing.

`Caused by: java.nio.file.NoSuchFileException: /home/ilippert/Documents/Studies/Publications&Products/Resources/Literature-bibtex.bib.tmp -> /home/ilippert/Documents/Studies/Publications&Products/Resources/Literature-bibtex.bib `

Can you manually check if after hitting save, a bib.tmp file is created

The Autocomplete error, however seems to be fixed.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -56,7 +56,7 @@ Note that this project **does not** adhere to [Semantic Versioning](http://semve
 - We fixed an issue where the Preferences entry preview will be unexpected modified leads to Value too long exception [#6198](https://github.com/JabRef/jabref/issues/6198)
 - We fixed an issue where custom jstyles for Open/LibreOffice would only be valid if a layout line for the entry type `default` was at the end of the layout section [#6303](https://github.com/JabRef/jabref/issues/6303)
 - We fixed an issue where sort on numeric cases was broken. [#6349](https://github.com/JabRef/jabref/issues/6349)
-
+- We fixed an issue where an "Not on FX thread" exception occured when saving on linux [#6453](https://github.com/JabRef/jabref/issues/6453)
 
 ### Removed
 

--- a/build.gradle
+++ b/build.gradle
@@ -543,7 +543,7 @@ task deleteInstallerTemp(type: Delete) {
 
 jpackage.dependsOn deleteInstallerTemp
 jlink {
-    options = ['--strip-debug','--compress', '2', '--no-header-files', '--no-man-pages']
+    options = ['--compress', '2', '--no-header-files', '--no-man-pages']
     launcher {
         name = 'JabRef'
     }

--- a/build.gradle
+++ b/build.gradle
@@ -543,7 +543,7 @@ task deleteInstallerTemp(type: Delete) {
 
 jpackage.dependsOn deleteInstallerTemp
 jlink {
-    options = ['--compress', '2', '--no-header-files', '--no-man-pages']
+    options = ['--strip-debug', '--compress', '2', '--no-header-files', '--no-man-pages']
     launcher {
         name = 'JabRef'
     }

--- a/src/main/java/org/jabref/gui/autocompleter/AutoCompletionTextInputBinding.java
+++ b/src/main/java/org/jabref/gui/autocompleter/AutoCompletionTextInputBinding.java
@@ -33,6 +33,8 @@ import javafx.scene.control.TextInputControl;
 import javafx.util.Callback;
 import javafx.util.StringConverter;
 
+import org.jabref.gui.util.DefaultTaskExecutor;
+
 import org.controlsfx.control.textfield.AutoCompletionBinding;
 
 /**
@@ -98,7 +100,7 @@ public class AutoCompletionTextInputBinding<T> extends AutoCompletionBinding<T> 
     }
 
     private static <T> StringConverter<T> defaultStringConverter() {
-        return new StringConverter<T>() {
+        return new StringConverter<>() {
             @Override
             public String toString(T t) {
                 return t == null ? null : t.toString();
@@ -133,7 +135,7 @@ public class AutoCompletionTextInputBinding<T> extends AutoCompletionBinding<T> 
             newText = "";
         }
         AutoCompletionInput input = inputAnalyzer.analyze(newText);
-        setUserInput(input.getUnfinishedPart());
+        DefaultTaskExecutor.runInJavaFXThread(() -> setUserInput(input.getUnfinishedPart()));
     }
 
     @Override

--- a/src/main/java/org/jabref/gui/fieldeditors/SimpleEditor.java
+++ b/src/main/java/org/jabref/gui/fieldeditors/SimpleEditor.java
@@ -33,10 +33,12 @@ public class SimpleEditor extends HBox implements FieldEditorFX {
         ((ContextMenuAddable) textInput).addToContextMenu(EditorMenus.getDefaultMenu(textInput));
         this.getChildren().add(textInput);
 
-        AutoCompletionTextInputBinding<?> autoCompleter = AutoCompletionTextInputBinding.autoComplete(textInput, viewModel::complete, viewModel.getAutoCompletionStrategy());
-        if (suggestionProvider instanceof ContentSelectorSuggestionProvider) {
-            // If content selector values are present, then we want to show the auto complete suggestions immediately on focus
-            autoCompleter.setShowOnFocus(true);
+        if (!isMultiLine) {
+            AutoCompletionTextInputBinding<?> autoCompleter = AutoCompletionTextInputBinding.autoComplete(textInput, viewModel::complete, viewModel.getAutoCompletionStrategy());
+            if (suggestionProvider instanceof ContentSelectorSuggestionProvider) {
+                // If content selector values are present, then we want to show the auto complete suggestions immediately on focus
+                autoCompleter.setShowOnFocus(true);
+            }
         }
 
         new EditorValidator(preferences).configureValidation(viewModel.getFieldValidator().getValidationStatus(), textInput);

